homeassistant/components/intelliclima/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""DataUpdateCoordinator for IntelliClima."""
+
+from datetime import timedelta
+
+from pyintelliclima import IntelliClimaAPI, IntelliClimaAPIError, IntelliClimaDevices
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, LOGGER
+
+type IntelliClimaConfigEntry = ConfigEntry[IntelliClimaCoordinator]
+
+
+class IntelliClimaCoordinator(DataUpdateCoordinator[IntelliClimaDevices]):
+    """Coordinator to manage fetching IntelliClima data."""
+
+    def __init__(
+        self, hass: HomeAssistant, entry: IntelliClimaConfigEntry, api: IntelliClimaAPI
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+            config_entry=entry,
+        )
+        self.api = api
+        self.data = IntelliClimaDevices.empty()
```

Comment:
Do we need this? I think we can save data to a local variable, and then coordinator will store it.